### PR TITLE
Add proposal instantiate 2

### LINKED ISSUE #1062: Add InstantiateContract2Proposal
Allow permission chains to use InstantiateContract2. PR to follow

### PR DESCRIPTION
**Pull request by GNaD13**
Close #1062

### CODE REVIEW

File: x/wasm/client/proposal_handler.go [hunk 1]
```diff
@@ -21,4 +21,5 @@ var ProposalHandlers = []govclient.ProposalHandler{
 	govclient.NewProposalHandler(cli.ProposalUnpinCodesCmd, rest.UnpinCodeProposalHandler),
 	govclient.NewProposalHandler(cli.ProposalUpdateInstantiateConfigCmd, rest.UpdateInstantiateConfigProposalHandler),
 	govclient.NewProposalHandler(cli.ProposalStoreAndInstantiateContractCmd, rest.EmptyRestHandler),
+	govclient.NewProposalHandler(cli.ProposalInstantiateContract2Cmd, rest.EmptyRestHandler),
```
Comment (alpe) [inline x/wasm/client/proposal_handler.go:24]: 👍 easy to miss

File: x/wasm/types/proposal.go [hunk 2]
```diff
@@ -34,6 +35,7 @@ var DisableAllProposals []ProposalType
 var EnableAllProposals = []ProposalType{
 	ProposalTypeStoreCode,
 	ProposalTypeInstantiateContract,
+	ProposalTypeInstantiateContract2,
```
Comment (alpe) [inline x/wasm/types/proposal.go:38]: 👍

File: x/wasm/types/authz.pb.go [hunk 3]
```diff
@@ -5,25 +5,21 @@ package types
 
 import (
 	fmt "fmt"
-	io "io"
```
Comment (alpe) [inline x/wasm/types/authz.pb.go:8]: Please run `make format` or `make proto-all` when you built from protobuf.

File: x/wasm/types/proposal_test.go [hunk 4]
```diff
@@ -279,6 +279,105 @@ func TestValidateInstantiateContractProposal(t *testing.T) {
...
+			}),
+			expErr: true,
+		},
+		"init funds with duplicates": {
+			src: InstantiateContract2ProposalFixture(func(p *InstantiateContract2Proposal) {
+				p.Funds = sdk.Coins{{Denom: "foo", Amount: sdk.NewInt(1)}, {Denom: "foo", Amount: sdk.NewInt(2)}}
+			}),
+			expErr: true,
+		},
+		"init with empty salt": {
```
Comment (alpe) [inline x/wasm/types/proposal_test.go:362]: 👍 good test cases

**Review by alpe — COMMENTED** [review]
👍 This looks great! Thanks a lot for the PR! 🥇 
If you would run `make format`, then you can get rid of the unrelated protobuf changes. 

I would like to to tag v0.30.0 final on main in the next days. I would keep your PR open to merge asap for v0.31

**Comment by GNaD13** [comment]
> +1 This looks great! Thanks a lot for the PR! 1st_place_medal If you would run `make format`, then you can get rid of the unrelated protobuf changes.
> 
> I would like to to tag v0.30.0 final on main in the next days. I would keep your PR open to merge asap for v0.31

Thank a lot for your reviews :3

**Pull request merged**